addon/components/bubble-chart/component.js

```diff
@@ -0,0 +1,168 @@
+import d3 from 'd3';
+import dc from 'dc';
+import crossfilter from 'crossfilter';
+import BaseChartComponent from '../base-chart-component';
+import moment from 'moment';
+/**
+   @public
+   @module bubble-chart
+   @type component
+   @desc dc.js bubble chart
+*/
+export default BaseChartComponent.extend({
+    classNames: ['bubble-chart'],
+
+    buildChart() {
+        let bubbleChart = dc.bubbleCloud(`#${this.get('elementId')}`);
+
+        let radii = [];
+        this.get('group').all().forEach(d => radii.push(this.getRadiusValue(d)));
+        let maxRadius = Math.max(...radii);
+
+        const titleFormatter = this.get('titleFormatter') || (value => value);
+
+        bubbleChart
+            .height(this.get('height'))
+            .dimension(this.get('dimension'))
+            .group(this.get('group'))
+            .x(d3.scale.ordinal())
+            .r(d3.scale.linear().domain([0, maxRadius * (this.get('group').size() / 4)]))
+            .radiusValueAccessor(d => this.getRadiusValue(d))
+            .label(d => titleFormatter(d.key))
+            .colors(d3.scale.quantize().domain([0, this.get('colors').length - 1]).range(this.get('colors')))
+            .colorAccessor(d => d.value.colorValue)
+            .renderTitle(false);
+
+        if (this.get('width')) {
+            bubbleChart.width(this.get('width'));
+        }
+
+        bubbleChart.on('renderlet', chart => this.onRenderlet(chart, this.createTooltip()));
+
+        this.set('chart', bubbleChart);
+    },
+
+    getRadiusValue(d) {
+        switch (this.get('radiusFormat')) {
+        case 'timestamp':
+            return moment.duration(moment().diff(moment(d.value.radius))).asMilliseconds();
+        case 'milliseconds':
+            return parseInt(d.value.radius);
+        case 'count':
+            return parseInt(d.value.radius);
+        default:
+            return parseInt(d.value.radius);
+        }
```

Comment:
This could probably just be

```
getRadiusValue(d) {
    if (this.get('radiusFormat') === 'timestamp') {
        return moment.duration(moment().diff(moment(d.value.radius))).asMilliseconds();
    }
    return parseInt(d.value.radius);
}
```

Reply:
I would normally agree and I see that it looks crazy to have this level of specificity, but I have it like this so that it's clear how we are getting the value based on the format specified. Idk if that's a good enough reason or not but if we add more format types in the future, or if (for example) the way we parse milliseconds changes, this seems the better way to go? That's my reasoning but if you still think it should be the shorter way, I'm cool with changing it

Reply:
Your call. I'm fine with leaving. My reasoning was that since any requirements change (like changing how we parse milliseconds) is going to require a code change anyway it wouldn't be hard to move to a switch statement down the road if needed. And until then we make the code as clean as possible.